client/src/components/RestaurantShow.js

```diff
@@ -28,36 +25,64 @@ const RestaurantShow = ({ currentUser, ...props }) => {
     }
   }
 
+  const addNewReview = async (formData) => {
+    try {
+      const response = await fetch('/api/v1/reviews', {
+        method: "POST",
+        headers: new Headers({
+          "Content-Type": "application/json"
+        }),
+        body: JSON.stringify({ formData })
+      })
+      if (!response.ok) {
+        if (response.status === 422) {
+          const errorBody = await response.json()
+          const newErrors = translateServerErrors(errorBody.errors)
+          return setErrors(newErrors)
+        }
+        throw new Error(`${response.status} ${response.statusText}`)
+      } else {
+        const body = await response.json()
+        const addedReview = body.review
+        setRestaurant({ ...restaurant, reviews: [...restaurant.reviews, addedReview] })
+        setReviewListOrForm("list")
+        return [true, addedReview];
+      }
+    } catch (error) {
+      console.error(`Fetch post error: ${error.name} ${error.message}`)
+    }
+  }
+
   const switchListOrForm = () => {
     switch (reviewListOrForm) {
       case "list":
         if (currentUser) {
           setReviewListOrForm("form")
         } else {
-          setErrors({user: "must be signed in to add a review"})
+          setErrors({ user: "must be signed in to add a review" })
         }
-        break;
+        break
       case "form":
         setReviewListOrForm("list")
-        break;
+        break
     }
-  } 
+  }
 
   useEffect(() => {
     getRestaurant()
-  }, [])
+  }, [currentUser])
```

Comment:
hmmm...why did we add this? we want to get the restaurant regardless of whether or not we have the user. remove this addition